### PR TITLE
Implemented static MapperConfiguration in BL.Mappers

### PR DESCRIPTION
**Pull request by Evgeniy-Sinicin**

### CODE REVIEW

File: InternshipBE/BL/Mappers/MapperConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using AutoMapper;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+
+namespace BL.Mappers
+{
+    public static class MapperConfiguration
+    {
+        public static AutoMapper.MapperConfiguration GetConfig(params Profile[] profiles)
```
Comment (romanstat) [inline InternshipBE/BL/Mappers/MapperConfiguration.cs]: remove parametrs

Comment (romanstat) [reply InternshipBE/BL/Mappers/MapperConfiguration.cs]: and change implementation

File: InternshipBE/BL/Mappers/MapperConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        {
+            return new AutoMapper.MapperConfiguration(mc => Array.ForEach(profiles, profile => mc.AddProfile(profile)));
+        }
+
+        public static IServiceCollection AddAutoMapper(this IServiceCollection services, IMapper mapper)
```
Comment (romanstat) [inline InternshipBE/BL/Mappers/MapperConfiguration.cs]: remove IMapper from parametr

File: InternshipBE/BL/Mappers/MapperConfigurationProvider.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            var configuration = new MapperConfiguration(Expression);
+            
+            configuration.AssertConfigurationIsValid();
+
+            return configuration;
+        }
+
+        public static IServiceCollection AddAutoMapper(this IServiceCollection services)
+        {
+            services.AddSingleton(GetConfig().CreateMapper());
```
Comment (romanstat) [inline InternshipBE/BL/Mappers/MapperConfigurationProvider.cs]: var mapper = GetConfig().CreateMapper();

return services.AddSingleton(mapper);

File: InternshipBE/BL/Mappers/MapperConfigurationProvider.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
+﻿using AutoMapper;
+using AutoMapper.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using System.Reflection;
+
+namespace BL.Mappers
```
Comment (romanstat) [inline InternshipBE/BL/Mappers/MapperConfigurationProvider.cs]: change folder name to Mapping and add folder Profiles

File: InternshipBE/BL/Mappers/MapperConfigurationProvider.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+{
+    public static class MapperConfigurationProvider
+    {
+        public static MapperConfigurationExpression Expression { get; }
```
Comment (romanstat) [inline InternshipBE/BL/Mappers/MapperConfigurationProvider.cs]: rename to MapperConfigurationExpression

File: InternshipBE/BL/Mappers/MapperConfigurationProvider.cs (hunk 3 above)
Comment (Evgeniy-Sinicin) [reply InternshipBE/BL/Mappers/MapperConfigurationProvider.cs]: Can I ask why?)
It is way to make code more readable?

Comment (romanstat) [reply InternshipBE/BL/Mappers/MapperConfigurationProvider.cs]: yea

File: InternshipBE/BL/Mappers/MapperConfigurationProvider.cs (hunk 4 above)
Comment (Evgeniy-Sinicin) [reply InternshipBE/BL/Mappers/MapperConfigurationProvider.cs]: Wow, Git doesn't see empty folder without .gitkeep file inside. I didn't know it :)

File: InternshipBE/BL/Mapping/MapperConfigurationProvider.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            configuration.AssertConfigurationIsValid();
+
+            return configuration;
+        }
+
+        public static IServiceCollection AddAutoMapper(this IServiceCollection services)
+        {
+            var mapper = GetConfig().CreateMapper();
+
+            services.AddSingleton(mapper);
```
Comment (romanstat) [inline InternshipBE/BL/Mapping/MapperConfigurationProvider.cs]: return

**Review by romanstat — APPROVED** [review]

**Review by vldmrklnk — APPROVED** [review]

**Pull request merged**